PINCache/PINOperationQueue.m

```diff
@@ -142,7 +133,27 @@ - (void)dealloc
 - (void)cancelOperation:(id <PINOperationReference>)operationReference
 {
   [self lock];
-    [_canceledOperations addObject:operationReference];
+    PINOperation *operation = [_referenceToOperations objectForKey:operationReference];
+    if (operation) {
+      for (NSMutableOrderedSet *queue in _prioritizedOperationQueues) {
+        [queue removeObject:operation];
```

Comment:
Either 66% or 100% of these `removeObject:` calls are going to "fall". This should be a cheap operation though, right?

The alternative would be storing the priority on the operation itself and using that to look up its queue. That doesn't seem worth it unless this iteration is costly.